c-sharp-razor/src/Minitwit.Infrastructure/Repository/AuthorRepository.cs

```diff
@@ -0,0 +1,269 @@
+using Microsoft.EntityFrameworkCore;
+using Minitwit.Core.Entities;
+using Minitwit.Core.Repository;
+
+namespace Minitwit.Infrastructure.Repository;
+
+public class AuthorRepository : BaseRepository, IAuthorRepository
+{
+    private readonly IFollowRepository _followRepository;
+
+    public AuthorRepository(MinitwitDbContext minitwitDbContext, IFollowRepository followRepository)
+        : base(minitwitDbContext)
+    {
+        _followRepository = followRepository;
+    }
+
+    // ----- Add Author Methods ----- //
+    public async void AddAuthorAsync(Author author)
+    {
+        await db.Users.AddAsync(author);
+        await db.SaveChangesAsync();
+    }
+
+    // ----- Get Author Methods ----- //
+
+    public async Task<ICollection<Author>> GetAllAuthorsAsync()
+    {
+        return await db.Users.ToListAsync();
+    }
+
+    public async Task<ICollection<Author>> GetAuthorsByIdAsync(IEnumerable<int> authorIds)
+    {
+        return await db.Users.Where(a => authorIds.Contains(a.Id)).AsNoTracking().ToListAsync();
+    }
+
+    public async Task<Author?> GetAuthorByIdAsync(int authorId)
+    {
+        Author? author = await db.Users.FirstOrDefaultAsync(a => a.Id == authorId);
+        return author!;
+    }
+
+    public async Task<Author> GetAuthorByNameAsync(string name)
+    {
+        Author? author = await db.Users.FirstOrDefaultAsync(a => a.UserName == name)!;
+        return author!;
+    }
+
+    public async Task<Author> GetAuthorByEmail(string email)
+    {
+        Author? author = await db.Users.FirstOrDefaultAsync(a => a.Email == email)!;
+        return author!;
+    }
+
+    // ----- Get Messages By Author and Page Methods ----- //
+    public async Task<ICollection<Message>> GetMessagesByAuthorAsync(int id)
```

Comment:
Same here, I see the signature has no page. But both functions are being used? GetMessagesByAuthorAsync x GetMessagesByAuthor. Why not call both the same then since both are async! And the signature of one requires id and the other one requires id, and page!